### ISSUE TITLE
lxqt-globalkeys pipe_utils.cpp FTBFS in hurd

### ISSUE THREAD

**Issue by agaida** [lxde/lxqt#926]
Maybe casting the integers is not the best idea, but with the casted values i prevent build time bugs in hurd-i386. The other architectures/platforms compile fine after the changes (linux/kfreebsd, amd64, arm64, mips  and so on). Can we do i so or are there better solutions?
@palinek @paulolieuthier 

<pre>
Description: cast return values to error_t
Author: Alf Gaida <agaida@siduction.org>

Bug: <url in upstream bugtracker>
Last-Update: 2015-12-27

--- lxqt-globalkeys-0.10.0.orig/daemon/pipe_utils.cpp
+++ lxqt-globalkeys-0.10.0/daemon/pipe_utils.cpp
@@ -39,10 +39,10 @@ void initBothPipeEnds(int fd[2])
 
 error_t createPipe(int fd[2])
 {
-    error_t result = 0;
+    error_t result = (error_t) 0;
     if (pipe(fd) < 0)
     {
-        result = errno;
+        result = (error_t) errno;
     }
     if (!result)
     {
@@ -59,16 +59,16 @@ error_t readAll(int fd, void *data, size
         ssize_t bytes_read = read(fd, data, length);
         if (bytes_read < 0)
         {
-            return errno;
+            return (error_t) errno;
         }
         if (!bytes_read)
         {
-            return -1;
+            return (error_t) -1;
         }
         data = reinterpret_cast<char *>(data) + bytes_read;
         length -= bytes_read;
     }
-    return 0;
+    return (error_t) 0;
 }
 
 error_t writeAll(int fd, const void *data, size_t length)
@@ -78,16 +78,16 @@ error_t writeAll(int fd, const void *dat
         ssize_t bytes_written = write(fd, data, length);
         if (bytes_written < 0)
         {
-            return errno;
+            return (error_t) errno;
         }
         if (!bytes_written)
         {
-            return -1;
+            return (error_t) -1;
         }
         data = reinterpret_cast<const char *>(data) + bytes_written;
         length -= bytes_written;
     }
-    return 0;
+    return (error_t) 0;
 }
 
 void closeBothPipeEnds(int fd[2])
</pre>

**Comment by paulolieuthier** [comment]
Extracted from the end of `/usr/include/errno.h`:
```
/* The Hurd <bits/errno.h> defines `error_t' as an enumerated type so
   that printing `error_t' values in the debugger shows the names.  We
   might need this definition sometimes even if this file was included
   before.  */
#if defined __USE_GNU || defined __need_error_t
# ifndef __error_t_defined
typedef int error_t;
#  define __error_t_defined	1
# endif
# undef __need_error_t
#endif
```
If `__error_t_defined` is not defined, it'll be in [`lxqt-globalkeys/daemon/pipe_utils.h`](https://github.com/lxde/lxqt-globalkeys/blob/master/daemon/pipe_utils.h#L34).

So, it'll be interesting to see how Hurd defines the enumeration in "`<bits/errno.h>`". Anyway, the casts are probably harmless, I'd just suggest using C++'s [`static_cast`](http://en.cppreference.com/w/cpp/language/static_cast) instead of plain C casts.

**Comment by paulolieuthier** [comment]
Here it is: https://sources.debian.net/src/glibc/2.21-6/sysdeps/mach/hurd/bits/errno.h/#L316.

**Comment by agaida** [comment]
@paulolieuthier no, it produces a FTBFS:
<pre>/«PKGBUILDDIR»/daemon/pipe_utils.cpp:42:22: error: invalid conversion from 'int' to 'error_t {aka __error_t_codes}' [-fpermissive]
     error_t result = 0;
                      ^</pre>
damn, i should learn what a static cast is :p

**Comment by agaida** [comment]
i've switched to static_cast and uploaded the package with this change to sid - if it build well i write a PR

**Comment by palinek** [comment]
fixed by lxde/lxqt-globalkeys@3001bee
